mux.go

```diff
@@ -209,6 +191,17 @@ func (r *Router) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		handler = http.NotFoundHandler()
 	}
 
+	if match.OnlyMatchedCleanPath {
+		// Added 3 lines (Philip Schlump) - It was dropping the query string and #whatever from query.
+		// This matches with fix in go 1.2 r.c. 4 for same problem.  Go Issue:
+		// http://code.google.com/p/go/issues/detail?id=5252
+		url := *req.URL
+		url.Path = match.CleanPath
+		w.Header().Set("Location", url.String())
+		w.WriteHeader(http.StatusMovedPermanently)
```

Comment:
```suggestion
		http.Redirect(w, req, url.String(), http.StatusMovedPermanently
```
based on @JoelSpeed 's suggestion